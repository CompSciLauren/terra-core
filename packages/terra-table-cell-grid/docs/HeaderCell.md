# Header Cell

The Terra TableCellGrid Header Cell is a sub component of the table header. The header cell dictates the minimum width of the column, but it can also dictate the maximum width of the column if a fixed or max-width value is set it.

While sort indicators are presented by the header cell, the actual sorting logic needs to be handled by the consumer by implementing the `onSelect` and `metaData` props.

## Usage

```jsx
import React from 'react';
import { HeaderCell } from 'terra-table-cell-grid';

<HeaderCell sort="asc" minWidth="small">
  Column 0
</HeaderCell>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
