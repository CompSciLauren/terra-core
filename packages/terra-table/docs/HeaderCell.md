# Header Cell

The Terra Table Header Cell is a sub component of the table header.

While sort indicators are presented by the header cell, the actual sorting logic needs to be handled by the consumer by implementing the `onSelect` and `metaData` props.

## Usage

```jsx
import React from 'react';
import { HeaderCell } from 'terra-table';

<HeaderCell sort="asc">
  Column 0
</HeaderCell>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
