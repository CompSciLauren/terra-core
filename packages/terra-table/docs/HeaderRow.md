# Header Row

The HeaderRow is a sub component that groups header cells passed through children. Can be set to synchronize space with Rows using check mark or chevron style selection.

## Usage

```jsx
import React from 'react';
import { HeaderRow, HeaderCell } from 'terra-table';

  <HeaderRow
    key="unique-react-key"
  >
    <HeaderCell>
      Cell Content
    </HeaderCell>
    <HeaderCell>
      Cell Content
    </HeaderCell>
  </HeaderRow>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
