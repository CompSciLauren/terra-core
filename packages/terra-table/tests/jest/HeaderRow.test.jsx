import React from 'react';
import HeaderRow from '../../src/HeaderRow';
import HeaderCell from '../../src/HeaderCell';

// Snapshot test
it('should render a default header cell grid', () => {
  const shallowComponent = shallow(<HeaderRow />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a cell grid with a cell', () => {
  const shallowComponent = shallow(<HeaderRow><HeaderCell /></HeaderRow>);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a checkmark selectionStyle cell grid', () => {
  const shallowComponent = shallow(<HeaderRow selectionStyle="checkmark" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a chevron selectionStyle cell grid', () => {
  const shallowComponent = shallow(<HeaderRow selectionStyle="chevron" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a cell grid with refCallback', () => {
  const shallowComponent = shallow(
    <HeaderRow refCallback={jest.fn()} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});
