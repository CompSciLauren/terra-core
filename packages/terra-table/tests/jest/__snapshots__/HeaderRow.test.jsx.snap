// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a checkmark selectionStyle row 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderCheckMarkCell
      isDisabled={false}
      isIntermediate={false}
      isPadded={true}
      isSelectable={false}
      isSelected={false}
    />
  </div>
</div>
`;

exports[`should render a chevron selectionStyle row 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderChevronCell
      isPadded={true}
    />
  </div>
</div>
`;

exports[`should render a default header row 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  />
</div>
`;

exports[`should render a row with a cell 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  >
    <HeaderCell
      isSelectable={false}
      sort="none"
    />
  </div>
</div>
`;

exports[`should render a row with refCallback 1`] = `
<div
  className="header"
  role="rowgroup"
>
  <div
    className="header-content"
    role="row"
  />
</div>
`;
